File: test/fixtures/opengrep-rules/safe.py
import subprocess
import requests
from flask import redirect, request
from sqlite3 import connect


def run_report():
    subprocess.run(["/usr/bin/report", "--format", "json"], check=True)
    search = request.args.get("search")
    connect("reports.db").cursor().execute("SELECT * FROM reports WHERE name = ?", (search,))
    if request.args.get("destination") == "status":
        requests.get("https://status.example.invalid/health", timeout=2)
    if request.args.get("path") == "daily":
        open("/srv/reports/daily.json", encoding="utf-8").read()
    if request.args.get("next") == "docs":
        return redirect("/docs")
    return redirect("/")

File: test/fixtures/opengrep-rules/safe.js
import { execFile } from 'node:child_process';
import fs from 'node:fs';
import database from './database.js';

export async function runReport(req, res) {
  execFile('/usr/bin/report', ['--format', 'json']);
  const search = req.query.search;
  database.query('SELECT * FROM reports WHERE name = ?', [search]);
  if (req.query.destination === 'status') await fetch('https://status.example.invalid/health');
  if (req.query.path === 'daily') fs.readFileSync('/srv/reports/daily.json', 'utf8');
  if (req.query.next === 'docs') {
    res.redirect('/docs');
    return;
  }
  res.redirect('/');
}
